File: app/src/main/java/com/bth/running/location/RunningBroker.java
package com.bth.running.location;

import com.bth.running.util.RunUpdate;

/**
 * @author Martin Macheiner
 *         Date: 03.10.2017.
 */

public interface RunningBroker {

    void startRun(long startMillis);

    void stopRun();

    interface RunningBrokerClient {

        void onRunUpdates(RunUpdate update);

        void onRunFinished();
    }

}

File: app/src/main/java/at/shockbytes/corey/running/location/RunningBroker.kt
package at.shockbytes.corey.running.location


import at.shockbytes.corey.running.util.RunUpdate

/**
 * @author  Martin Macheiner
 * Date:    03.10.2017.
 */

interface RunningBroker {

    fun startRun(startMillis: Long)

    fun stopRun()

    interface RunningBrokerClient {

        fun onRunUpdates(update: RunUpdate)

        fun onRunFinished()
    }

}
